api/src/main/java/io/druid/timeline/partition/ShardSpec.java

```diff
@@ -41,4 +43,6 @@
   public int getPartitionNum();
 
   public ShardSpecLookup getLookup(List<ShardSpec> shardSpecs);
+
+  public Map<String, RangeSet<String>> getDomain();
```

Comment:
Please add a javadoc for this method, it's not super clear from the signature what it should mean.